tests/pipeline/test_quarters_estimates.py

```diff
@@ -0,0 +1,747 @@
+        pass
+
+    @classmethod
+    def init_class_fixtures(cls):
+        cls.sids = cls.events['sid'].unique()
+        cls.columns = {
+            Estimates.estimate: 'estimate',
+            Estimates.event_date: EVENT_DATE_FIELD_NAME,
+            Estimates.fiscal_quarter: FISCAL_QUARTER_FIELD_NAME,
+            Estimates.fiscal_year: FISCAL_YEAR_FIELD_NAME,
+        }
+        cls.loader = cls.make_loader(cls.events, cls.columns)
+
+        cls.ASSET_FINDER_EQUITY_SIDS = list(
+            cls.events[SID_FIELD_NAME].unique()
+        )
+        cls.ASSET_FINDER_EQUITY_SYMBOLS = [
+            's' + str(n) for n in cls.ASSET_FINDER_EQUITY_SIDS
+        ]
+        super(EstimateTestCase, cls).init_class_fixtures()
+
+    def _test_wrong_num_quarters_passed(self):
+        with self.assertRaises(ValueError):
+            dataset = QuartersEstimates(-1)
+            engine = SimplePipelineEngine(
+                lambda x: self.loader,
+                self.trading_days,
+                self.asset_finder,
+            )
+
+            engine.run_pipeline(
+                Pipeline({c.name: c.latest for c in dataset.columns}),
+                start_date=self.trading_days[0],
+                end_date=self.trading_days[-1],
+            )
+
+
+window_test_cases = [
+    (window_len, start_idx, num_quarters_out) for
+    (window_len, start_idx), num_quarters_out in
+    itertools.product(
+        [[5, pd.Timestamp('2015-01-09').tz_localize('utc')],
+         [6, pd.Timestamp('2015-01-12').tz_localize('utc')],
+         [11, pd.Timestamp('2015-01-20').tz_localize('utc')],
+         [19, pd.Timestamp('2015-01-30').tz_localize('utc')],
+         [26, pd.Timestamp('2015-02-10').tz_localize('utc')]],
+        [1, 2, 3, 4])
+]
+
+
+class NextEstimateWindowsTestCase(EstimateTestCase,
+                                  WithAssetFinder,
+                                  WithTradingSessions,
+                                  ZiplineTestCase):
+    START_DATE = pd.Timestamp('2014-12-31')
+    END_DATE = pd.Timestamp('2015-02-15')
+    events = estimates_timeline
+
+    @classmethod
+    def make_loader(cls, events, columns):
+        return NextQuartersEstimatesLoader(events, columns)
+
+    @parameterized.expand(window_test_cases)
+    def test_next_estimate_windows_at_quarter_boundaries(self,
+                                                         window_len,
+                                                         start_idx,
+                                                         num_quarters_out):
+        """
+        Tests that we overwrite values with the correct quarter's estimate at
+        the correct dates.
+        """
+        dataset = QuartersEstimates(num_quarters_out)
+
+        class SomeFactor(CustomFactor):
+            inputs = [dataset.estimate]
+            window_length = window_len
+
+            def compute(self, today, assets, out, *inputs):
+                unique_inputs = np.unique(inputs).tolist()
```

Comment:
I find this test extremely confusing. Could we just store the window for in a dictionary for each day and then walk through the days and assert that the windows are equal to some hard coded expectations? Also, we are only testing with constants for each earnings quarter so it may be misaligned.